LIPs/LIP-bond-event-details.md

```diff
@@ -0,0 +1,65 @@
+the delegate's address and the delegator's address. The `Bond` event should provide additional data fields that allow off-chain clients to respond to 
+more specific user actions with the contract and to reconstruct intermediate contract states resulting from past state transitions for the 
+execution of the `bond()` function.
+
+## Motivation
+
+Off-chain clients commonly use contract events to respond to certain user actions with a contract and to reconstruct intermediate contract states that 
+can be used to create history for the contract's state which otherwise would not be available without the use of an Ethereum archival node to explicitly 
+trace contract state history. 
+
+An example of a Livepeer protocol related application that would like to leverage the capabilities described above is [Supermax](https://www.supermax.cool/livepeer). 
+The Supermax team would like accurately [track stake changes](https://github.com/livepeer/protocol/issues/224) of transcoders based on the delegation activity of delegators within their application. At the 
+moment this feature is impossible because the only way to track stake changes of transcoders is via the `Bond` event in the BondingManager, but it does not 
+contain data about the transcoder that a delegator was previously delegated to when a delegator changes its delegate from transcoder A to transcoder B. Including 
+data fields for a delegator's old delegate in addition to how a delegator's bonded amount changes when the `bond()` function is invoked can help 
+not only make this specific feature possible, but open up a larger design space for off-chain clients that might want to make use of this information.
+
+## Specification
+
+Update the `Bond` event signature in the BondingManager contract to:
+
+```
+Bond(address indexed newDelegate, address indexed oldDelegate, address indexed delegator, uint256 additionalAmount, uint256 delegationAmount);
+```
+
+The fields in the above event signature are:
+- `newDelegate`: The address of the delegator's new delegate. If the delegator is not changing its existing delegate (not the null address), then 
+this should be the address of its existing delegate.
+- `oldDelegate`: The address of the delegator's old delegate. If the delegator did not have an existing delegate (not the null address), then
+this should be the null address. If the delegator did have an existing delegate which it is not changing, then this should be the address of its existing delegate.
+- `additionalAmount`: The amount of additional tokens being bonded. If the delegator is not bonding additional tokens, this should be 0.
+- `delegationAmount`: The amount of bonded tokens being delegated. This value should always reflect the delegator's current bonded amount. If `additionalAmount > 0`, 
+this value should be the sum of the delegator's bonded amount before the current execution of the `bond()` function and `additionalAmount`.
```

Comment:
Keep in mind that the `bond()` txn also updates their delegated amount based upon unclaimed rewards, so maybe that's worth a mention here if it's not semantically correct.

Reply:
Yeah good point - I'll clarify that the delegator's current bonded amount will also reflect claimed rewards added to the delegator's bonded amount right before the execution of the main logic for `bond()` via the `autoClaimEarnings` modifier.